rust/src/measurements/gumbel_max/make_report_noisy_max_gumbel.tex

```diff
@@ -1,253 +1,248 @@
+The $i^{th}$ element of each score vector is the score for the $i^{th}$ candidate. 
+The sensitivity of the scoring function can be measured in terms of the $L_\infty$ norm, which we name the \texttt{LInfDistance}.  
+It characterizes the greatest that any one score may change: 
+\begin{equation} 
+    \Delta_{\infty} = \max_{x \sim x'} d_{\infty}(f(x), f(x')) = \max_{x \sim x'} \max_i \abs{f(x)_i - f(x')_i} 
+\end{equation} 
+Unfortunately, this choice of metric always results in a loosening by a factor of 2 when evaluating the privacy guarantee of the exponential mechanism. 
+This is because both the $i^{th}$ likelihood and normalization term may vary in opposite directions, resulting in a more distinguishing event. 
+However, this loosening is not necessary if we can prove that the scoring function is monotonic, because the $i^{th}$ likelihood and normalization term will always vary in the same direction. 
+ 
+We instead use a slight adjustment to this metric, \texttt{RangeDistance}, characterizing the greatest difference in scores: 
+\begin{equation} 
+    \Delta_{\text{Range}} = \max_{x \sim x'} d_{\text{Range}}(f(x), f(x')) = \max_{x \sim x'} \max_{ij} \abs{(f(x)_i - f(x')_i) - (f(x)_j - f(x')_j)} 
+\end{equation} 
+Consider when the scoring function is not monotonic. 
+The sensitivity is maximized when $x_i - x'_i$ and $x_j - x'_j$ vary maximally in opposite directions, resulting in the same loosening factor of 2. 
+On the other hand, when the scoring function is monotonic, the sign of the $x_i - x'_i$ term matches the sign of the $x_j - x'_j$ term, 
+and their magnitudes cancel. 
+Therefore, when the scorer is monotonic, the sensitivity is maximized when one term is zero.  
+It is shown in \ref{privacy-guarantee} that a tighter analysis of the exponential mechanism is compatible with a score vector whose sensitivity is expressed in terms of this metric. 
+ 
+Given that both the infinity-distance and range-distance are useful, the mechanism still uses the infinity-distance, 
+but an additional boolean is stored in the metric to indicate when the score is monotonic. 
+ 
+ 
+\section{Hoare Triple} 
+\subsection*{Precondition} 
+\textbf{Compiler-verified}
+\begin{itemize} 
+    \item \texttt{TIA} (atomic input type) is a type with trait \rustdoc{traits/trait}{Number}
+    \item \texttt{f64} is a type with trait \rustdoc{traits/trait}{DistanceConstant}\texttt{<TIA>}
+    \item \texttt{FBig} is a type that supports fallible exact casting from \texttt{TIA} and \texttt{f64}
+\end{itemize} 
+
+\textbf{Human-verified}
+None
+ 
+\subsection*{Function} 
+\label{sec:python-pseudocode} 
+\lstinputlisting[language=Python,firstline=2]{./pseudocode/make_report_noisy_max_gumbel.py} 
+ 
+\subsection*{Postcondition} 
+ 
+\validMeasurement{\texttt{input\_domain, input\_metric, scale, optimize, TIA}}{\texttt{make\_report\_noisy\_max\_gumbel}}  
+ 
+\section{Proof} 
+\subsection{Privacy Guarantee} 
```

Comment:
Do we really need a subsection when there is only one? Can we just change the name of the section to, "Proof of Privacy", to make it cleaner?

Reply:
I've recently added a second condition for data-independent runtime errors. :/